sphinxgallery/gen_rst.py

```diff
@@ -444,7 +458,12 @@ def generate_file_rst(fname, target_dir, src_dir):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
+    exec_script = fname.startswith('plot')
+    use_mayavi = gallery_conf.get('use_mayavi', False)
+    if not use_mayavi and any('mayavi' in bl[1] for bl in script_blocks):
```

Comment:
> basically it says that we don't want to execute an example that uses mayavi if
> use_mayavi is False. Otherwise the demo gallery does not build fine if mayavi
> is not present. Makes sense?

Add this in a comment.